clang/lib/Driver/ToolChains/AIX.cpp

```diff
@@ -127,9 +127,19 @@ void aix::Linker::ConstructJob(Compilation &C, const JobAction &JA,
   }
 
   // Force static linking when "-static" is present.
-  if (Args.hasArg(options::OPT_static))
+  if (Args.hasArg(options::OPT_static)) {
     CmdArgs.push_back("-bnso");
 
+    if (D.IsFlangMode()) {
+      // The folllowing linker options are needed to statically link to the
+      // shared libflang_rt.runtime.a on AIX
+      CmdArgs.push_back("-bI:/usr/lib/syscalls.exp");
+      CmdArgs.push_back("-bI:/usr/lib/aio.exp");
```

Comment:
Adding this helps to resolve none of the symbols needed when `libflang_rt.runtime.so` is relinked (with the same set of exports) using `-bnso -lc -lpthreads -lcrypt`.
```suggestion
```